File: hashmap/1_Two_Sum.py
# time: O(n)
# space: O(n)

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for i, value in enumerate(nums):
            remaining = target - value
            if remaining in hashmap:
                return [i, hashmap[remaining]]
            else:
                hashmap[value] = i

File: Array/1_Two_Sum.go
// time: O(n)
//space: O(n)

func twoSum(nums []int, target int) []int {
	numMap := make(map[int]int)
	for i, num := range nums {
	 complement := target - num
	 index, exists := numMap[complement]
	 if exists {
			 return []int {i, index}
	 }
	 numMap[num] = i
	} 
	return nil
}